pkg/users/account-create-dialog.js

```diff
@@ -222,57 +301,60 @@
-                    return cockpit.spawn(prog, { superuser: "require", err: "message" })
-                            .then(() => passwd_change(user_name, password))
-                            .then(() => {
-                                if (locked)
-                                    return cockpit.spawn([
-                                        "/usr/sbin/usermod",
-                                        user_name,
-                                        "--lock"
-                                    ], { superuser: "require", err: "message" });
-                                if (force_change)
-                                    return cockpit.spawn([
-                                        "/usr/bin/passwd",
-                                        "-e",
-                                        user_name
-                                    ], { superuser: "require", err: "message" });
-                            });
+    function create(real_name, user_name, password, locked, force_change, home_dir, base_home_dir) {
+        const prog = ["/usr/sbin/useradd", "--create-home", "-s", state.shell];
+        if (real_name) {
+            prog.push('-c');
+            prog.push(real_name);
+        }
+
+        if (home_dir) {
+            prog.push('-d');
+            prog.push(home_dir);
+        }
+        prog.push(user_name);
+        return cockpit.spawn(prog, { superuser: "require", err: "message" })
+                .then(() => passwd_change(user_name, password))
+                .then(() => {
+                    if (locked)
+                        return cockpit.spawn([
+                            "/usr/sbin/usermod",
+                            user_name,
+                            "--lock"
+                        ], { superuser: "require", err: "message" });
+                    if (force_change)
+                        return cockpit.spawn([
+                            "/usr/bin/passwd",
+                            "-e",
+                            user_name
+                        ], { superuser: "require", err: "message" });
                 });
     }
 
-    function passwd_check(force_weak, real_name, user_name, password, password_confirm, locked, force_change) {
+    function passwd_check(force_weak, real_name, user_name, password, password_confirm, locked, home_dir, base_home_dir, force_change) {
         return validate(force_weak, real_name, user_name, password, password_confirm).then(valid => {
             if (valid)
-                return create(real_name, user_name, password, locked, force_change);
+                return create(real_name, user_name, password, locked, force_change, home_dir, base_home_dir);
             else {
-                if (!errors.real_name && !errors.user_name && !errors.password_confirm && state.password.length <= 256) {
+                if (!errors.real_name && !errors.user_name && !errors.home_dir && !errors.password_confirm && state.password.length <= 256) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18234-20230131-135835-5b43f8e5-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)